mobile/src/store/courses/actions.ts

```diff
@@ -74,6 +71,102 @@ const getMentorsByCourseId = createAsyncThunk<
   return mentors;
 });
 
+const updateVisibilityBecomeMentor = createAsyncThunk<
+  boolean,
+  void,
+  AsyncThunkConfig
+>(ActionType.UPDATE_VISIBILITY_BECOME_MENTOR, async (_, { getState }) => {
+  const {
+    auth: { user },
+    courses: { mentors, course },
+  } = getState();
+  const isMentorBecomingVisible =
+    course?.courseCategoryId &&
+    !mentors.some((mentor) => mentor.id === user?.id);
+
+  return Boolean(isMentorBecomingVisible);
+});
+
+const setBecomeMentorInvisible = createAsyncThunk<
+  boolean,
+  void,
+  AsyncThunkConfig
+>(ActionType.SET_BECOME_MENTOR_INVISIBLE, async () => {
+  return false;
+});
+
+const createMentor = createAsyncThunk<
+  CoursesToMentorsResponseDto,
+  CoursesToMentorsRequestDto,
+  AsyncThunkConfig
+>(ActionType.CREATE_MENTOR, async (payload, { extra }) => {
+  const { mentorsApi } = extra;
+
+  return mentorsApi.create(payload);
+});
+
+const becomeMentor = createAsyncThunk<void, void, AsyncThunkConfig>(
+  ActionType.BECOME_MENTOR,
+  async (_, { dispatch, getState, extra }) => {
+    const {
+      courses: { course },
+      auth: { user },
+    } = getState();
+
+    const { interviewsApi } = extra;
+
+    if (!user || !course) {
+      return;
+    }
+
+    const passedCategoryInterviews =
+      await interviewsApi.getPassedInterviewCategoryIds(user.id);
+    const isCategoryPassed = passedCategoryInterviews.includes(
+      course.courseCategoryId,
+    );
+
+    if (isCategoryPassed) {
+      const payload: CoursesToMentorsRequestDto = {
+        userId: user.id,
+        courseId: course.id,
+      };
+
+      try {
```

Comment:
Why do we need try/catch here?

Reply:
I thought in order to handle the error and show a message about it

Reply:
or 
await dispatch(createMentor(payload)).unwrap(); 
would it be enough

Reply:
We have handle error middleware for this

Reply:
Remove try/catch block

Reply:
ok